### ISSUE TITLE
Unable to decode on query_map call while iterating a QueryMapResult

### ISSUE THREAD

**Issue by apuigsech** [polkascan/py-substrate-interface#99]
This code;

```
from substrateinterface import SubstrateInterface

s = SubstrateInterface(
        url='wss://kusama-rpc.polkadot.io/',
        type_registry_preset="kusama",
)

bonded_map = s.query_map('Staking', 'Bonded', page_size=100)

bonded = {}
for k,v in bonded_map:
	bonded[k.value] = v.value
```

produces the following error after calling 'retrieve_next_page' 5 times (If use page_size=500, it fails after calling only one).

```
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    for k,v in bonded_map:
  File "/usr/local/lib/python3.8/site-packages/substrateinterface/base.py", line 3247, in __next__
    self.records += self.retrieve_next_page(start_key=self.last_key)
  File "/usr/local/lib/python3.8/site-packages/substrateinterface/base.py", line 3226, in retrieve_next_page
    result = self.substrate.query_map(module=self.module, storage_function=self.storage_function,
  File "/usr/local/lib/python3.8/site-packages/substrateinterface/base.py", line 1308, in query_map
    item_key = self.decode_scale(
  File "/usr/local/lib/python3.8/site-packages/substrateinterface/base.py", line 2656, in decode_scale
    obj.decode()
  File "/usr/local/lib/python3.8/site-packages/scalecodec/base.py", line 390, in decode
    raise RemainingScaleBytesNotEmptyException('Current offset: {} / length: {}'.format(self.data.offset, self.data.length))
scalecodec.exceptions.RemainingScaleBytesNotEmptyException: Current offset: 32 / length: 24
```

**Comment by apuigsech** [comment]
I dirty-editet the code to prevent it to stop on a decoding failure (by using try/except) and it is able to retrieve much more data. It looks there is a corner case somewhere.

This is the change I did:

```
                    **try:**
                        item_key = self.decode_scale(
                        type_string=key_type,
                        scale_bytes='0x' + item[0][len(prefix) + concat_hash_len:],
                        return_scale_obj=True,
                        block_hash=block_hash
                        )

                         # Automatic SS58 encode AccountId
                        if type(item_key) is GenericAccountId:
                                item_key.ss58_address = self.ss58_encode(item_key.value)

                        item_value = self.decode_scale(
                        type_string=value_type,
                        scale_bytes=item[1],
                        return_scale_obj=True,
                        block_hash=block_hash
                        )

                        # Automatic SS58 encode AccountId
                        if type(item_value) is GenericAccountId:
                                item_value.ss58_address = self.ss58_encode(item_value.value)

                        result.append([item_key, item_value])
                    **except:
                        None**

```

**Comment by arjanz** [comment]
I managed to reproduce this one, I will investigate the type registry mismatch

**Comment by arjanz** [comment]
Interesting case you have presented here.. It seems the key that results in a decoding error is not 32 bytes (as it should, because it's an `AccountID`, but this one is only 24 bytes. (raw data of key (minus prefix): `0x5a8c9bfe3df728c83660a5c3c5f9f7f89a5b897f50d53ffb`).

I'll have to investigate why this is possible and how to handle these exceptions. If for some reason it is possible to have invalid keys in the storage, I may have to build some way to ignore these exceptions. 

I see that Polkadot.JS just outputs the full raw key (`0x5f3e4907f716ac89b6347d15ececedca3ed14b45ed20d054f05e37e2542cfe7012b8dfc2b39bf2dd5a8c9bfe3df728c83660a5c3c5f9f7f89a5b897f50d53ffb`), you can reproduce this by inspecting the results at https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkusama-rpc.polkadot.io#/chainstate

**Comment by arjanz** [comment]
Included in release v0.12.6
